### DISCUSSION TITLE
CORS error when endpoint logic broken

### DISCUSSION THREAD

**Discussion by ScrimForever** [fastapi/fastapi#8835] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
--
```

### Description

When my code broken on endpoint, anyway becouse i did logic wrong, browser shows CORS error. I just can see whats wrong, on shell console. But in somecases i deploy it on AppEngine and you cant see logic error output easyly .

### Operating System

Linux

### FastAPI Version

0.68.1

### Python Version

3.8.10

**Comment by panla** [comment]
custom catch exception

example

```python
def register_exception(app: FastAPI):
    @app.exception_handler(BaseHTTPException)
    async def catch_c_http_exception(request: Request, exc: BaseHTTPException):
        """catch custom exception"""

        log_message(request, exc.message)
        content = {'status_code': exc.CODE, 'message': exc.message, 'data': None}
        return JSONResponse(content=content, status_code=exc.status_code, headers=exc.headers)

    @app.exception_handler(Exception)
    async def exception_handle(request: Request, exc: Exception):
        """catch other exception"""

        log_message(request, traceback.format_exc())
        content = {'status_code': StatusCode.server_error, 'message': str(exc), 'data': None}
        return JSONResponse(content=content, status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

**Comment by ScrimForever** [comment]
@panla , this not help me. I dont know what libs i need import to test it. Can you help me ?

**Comment by panla** [comment]
if you write some error/bug code (`print(1 / 0)`)  in your api. this can catch exception

another is cors middleware 

example <https://github.com/panla/fastapi_tortoise_mysql>
